src/DssProxyActions.sol

```diff
@@ -110,10 +110,14 @@ interface ProxyLike {
 contract Common {
     uint256 constant RAY = 10 ** 27;
 
-    VatLike immutable public vat;
+    VatLike     immutable public vat;
+    DaiJoinLike immutable public daiJoin;
+    GemLike     immutable public dai;
 
-    constructor(address vat_) public {
-        vat = VatLike(vat_);
+    constructor(address daiJoin_) public {
+        vat = VatLike(DaiJoinLike(daiJoin_).vat());
+        daiJoin = DaiJoinLike(daiJoin_);
```

Comment:
Neat - initiate `daiJoin` either before or after both of `vat` and `dai` (as they both read from it in the same manner).

Reply:
Done!